CLI: --list-different + --write = Exit status 1 if something was modified

It doesn't play very well with npm scripts:

```
d:\dev\foo>npm run format

> @ format d:\dev\foo
> prettier --write --list-different *.js

foo.js
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ format: `prettier --write --list-different *.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ format script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

Without `--list-different` though it doesn't behave like this.

--- a/src/cli/util.js
+++ b/src/cli/util.js
@@ -146,8 +146,8 @@ function listDifferent(context, input, options, filename) {
     if (!prettier.check(input, options)) {
       if (!context.argv["write"]) {
         context.logger.log(filename);
+        process.exitCode = 1;
       }
-      process.exitCode = 1;
     }
   } catch (error) {
     context.logger.error(error.message);
@@ -498,7 +498,9 @@ function formatFiles(context) {
 
     if (context.argv["list-different"] && isDifferent) {
       context.logger.log(filename);
-      process.exitCode = 1;
+      if (!context.argv["write"]) {
+        process.exitCode = 1;
+      }
     }
 
     if (context.argv["write"]) {

--- a/tests_integration/__tests__/infer-parser.js
+++ b/tests_integration/__tests__/infer-parser.js
@@ -113,7 +113,7 @@ describe("--write and --list-different with unknown path and no parser", () => {
 
   describe("multiple files", () => {
     runPrettier("cli/infer-parser/", ["--list-different", "--write", "*"]).test(
-      { status: 1 }
+      { status: 0 }
     );
   });
 });

--- a/tests_integration/__tests__/piped-output.js
+++ b/tests_integration/__tests__/piped-output.js
@@ -8,7 +8,7 @@ describe("output with --list-different + unformatted differs when piped", () => 
     ["--write", "--list-different", "--no-color", "unformatted.js"],
     { stdoutIsTTY: true }
   ).test({
-    status: 1
+    status: 0
   });
 
   const result1 = runPrettier(
@@ -16,7 +16,7 @@ describe("output with --list-different + unformatted differs when piped", () => 
     ["--write", "--list-different", "--no-color", "unformatted.js"],
     { stdoutIsTTY: false }
   ).test({
-    status: 1
+    status: 0
   });
 
   expect(result0.stdout.length).toBeGreaterThan(result1.stdout.length);

